vdyp-sindex/sindxdll.java

```diff
@@ -0,0 +1,3947 @@
+ *
+ *  Return Value
+ *  ------------
+ *    Integer code.
+ *    May return an error code under the following conditions:
+ *    
+ *    return value    condition
+ *    ------------    ---------
+ *    SI_ERR_CURVE    input curve is not a valid curve index.
+ *
+ *  Remarks
+ *  -------
+ *    Code bits are set as follows:
+ *      0001: ht = fn (si, age)
+ *      0010: si = fn (ht, age)
+ *      0100: y2bh = fn (si)
+ *      1000: si = fn (ht, age) growth intercept
+ -----------------------------------------------------------------------------*/
+public static short Sindex_CurveUse (short cu_index) {
+  if (cu_index >= 0 && cu_index < SI_MAX_CURVES){
+    return (short) (si_curve_types[cu_index]);
+  }
+  
+  return SI_ERR_CURVE;
+}
+
+
+//These all call other functions from other C files
+/*-----------------------------------------------------------------------------
+ *
+ *  Sindex_HtAgeToSI
+ *  ================
+ *    Converts a Height and Age to a Site Index for a particular Site Index
+ *    Curve.
+ *
+ *  Parameters
+ *  ----------
+ *    Integer curve index.
+ *      The particular site index curve to project the height and age along.
+ *
+ *    Floating point age.
+ *      The age of the trees indicated by the curve selection.  The
+ *      interpretation of this age is modified by the 'ageType' parameter.
+ *
+ *    Integer age type.
+ *      Must be one of:
+ *        SI_AT_TOTAL    The age is the total age of the stand in years since
+ *                planting.
+ *
+ *        SI_AT_BREAST  The age indicates the number of years since the stand
+ *                reached breast height.
+ *
+ *    Floating point height.
+ *      The height of the species in metres.
+ *
+ *    Integer estimate type.
+ *      Must be one of:
+ *        SI_EST_DIRECT    Compute the site index based on direct equations
+ *                    if available.  If the equations are not available,
+ *                    then automatically fall to the SI_EST_ITERATE
+ *                    method.
+ *
+ *        SI_EST_ITERATE    Compute the site index based on an iterative
+ *                    method which converges on the true site index.
+ *
+ *    Floating point site index. (computed)
+ *      This value is computed from the other parameters.
+ *      If an error condition occurs, the site index is set to the
+ *        same as the return value.
+ *
+ *  Return Value
+ *  ------------
+ *    0, or an error code under the following conditions:
+ *    
+ *    return value    condition
+ *    ------------    ---------
+ *    SI_ERR_CURVE    input curve is not a valid curve index.
+ *    SI_ERR_GI_MIN   if bhage < 0.5
+ *    SI_ERR_GI_MAX   if bhage > GI range
+ *    SI_ERR_NO_ANS   if computed SI > 999
+ *    SI_ERR_GI_TOT   if total age and GI curve
+ -----------------------------------------------------------------------------*/
+public static short Sindex_HtAgeToSI(
+    short curve, 
+    double age, 
+    short ageType, 
+    double height, 

```

Comment:
Just want to confirm that `double` (64bit floating point) is correct  for siindex?  The rest of VDYP is using `float` (32bit floating point)

Reply:
Double checked, they use double in the original C code.